go/vt/vttablet/endtoend/misc_test.go

```diff
@@ -759,3 +759,19 @@ func TestAppDebugRequest(t *testing.T) {
 		t.Errorf("Error: %v, want prefix %s", err, want)
 	}
 }
+
+func TestBeginExecuteWithFailingPreQueriesAndCheckConnectionState(t *testing.T) {
+	client := framework.NewClient()
+
+	insQuery := "insert into vitess_test (intval, floatval, charval, binval) values (4, null, null, null)"
+	preQueries := []string{
+		"savepoint a",
```

Comment:
I'm not seeing how you are asserting that the pre-queries actually ran. We could change the system session state, or maybe do the insert as a pre-query, and the select as the main query. WDYT?

Reply:
on L769 there is "release savepoint b" but there does not exists savepoint b so it will fail and I validate that the insert was not executed by doing select using same client. Also, I do not do any rollback or commit. So there is no transaction killer involved as well.